### PR TITLE
Jc/cors

### PR DESCRIPTION
**Pull request by joeconwaystk**
This PR:

1. Updates CORS tests and improves behavior according to the specification.
2. Changes behavior of unimplemented HTTP methods in HTTPController to return a 405 status code with Allow header instead of a 404.

### CODE REVIEW

File: test/base/cors_test.dart [hunk 1]
```diff
@@ -5,202 +5,477 @@ import 'package:http/http.dart' as http;
...
 
-      // Should return 401, the preflight is unsigned and therefore doesn't make it past the authenticator, since it
-      // won't allow OPTIONS thru because it isn't expecting a preflight request.
-      expect(resp.statusCode, 401);
+      expect(resp.statusCode, 200);
+      expect(resp.headers["access-control-allow-origin"], isNull);
+      expect(resp.headers["access-control-allow-headers"], isNull);
+      expect(resp.headers["access-control-allow-methods"], isNull);
+      expect(resp.headers["access-control-expose-headers"], isNull);
+      expect(resp.headers["access-control-allow-credentials"], isNull);
```
Comment (anachlas) [inline test/base/cors_test.dart]: Can use the convenience `expectThatNoCORSProcessingOccured` here.

Comment (joeconwaystk) [inline test/base/cors_test.dart]: Technically, I can't because these are an OPTIONS and it is a GET, but I can just make an OPTIONS version

Comment (joeconwaystk) [inline test/base/cors_test.dart]: gah, nvm, I forgot what that method did.

File: lib/http/router.dart [hunk 2]
```diff
@@ -19,6 +19,7 @@ class Router extends RequestController {
   /// Creates a new [Router].
   Router() {
     unhandledRequestController = _handleUnhandledRequest;
+    policy.allowCredentials = false;
```
Comment (JesseBlack82) [inline lib/http/router.dart]: Why do you override the CORS default here?

**Comment by JesseBlack82** [comment]
👍

**Pull request merged**